File: 2022/01/12/c/search_insert_position.c
// Program: https://leetcode.com/problems/search-insert-position/
// Author: Andrew Jarombek
// Date: 1/12/2022

int searchInsert(int* nums, int numsSize, int target){
    for (int i = 0; i < numsSize; i++) {
        if (nums[i] == target || nums[i] > target) {
            return i;
        }
    }

    return numsSize;
}

File: 2022/01/22/rust/src/search_insert_position.rs
// Program: https://leetcode.com/problems/search-insert-position/
// Author: Andrew Jarombek
// Date: 1/22/2022

pub fn search_insert(nums: Vec<i32>, target: i32) -> i32 {
    let mut i = 0;

    while i < nums.len() {
        if nums[i] >= target {
            return i as i32;
        }

        i += 1;
    }

    return i as i32;
}